mono/utils/mono-machine.h

```diff
@@ -20,6 +20,20 @@
 #include "config.h"
 #include <glib.h>
 
+#if defined (__x86_64__) || defined (__arm64__) || defined (__aarch64__)
+// Note: This is independent of __ILP32__ and __LP64__ and is not
+// the same thing as HOST_AMD64 and HOST_ARM64. It is more like
+// defined (HOST_X32) || defined (HOST_ARM6432).
```

Comment:
It might be the same as HOST_AMD64/ARM64, as X32/ARM6432 might also define those. It is a bit hypothetical right now but might be non-hypothetical soon.

For now we could just use gssize unconditionally here for host_mgreg_t.
We might also consider having unsigned  and signed registers.